eclipse/test/junit/saros/editor/internal/ContributionAnnotationHistoryTest.java

```diff
@@ -0,0 +1,86 @@
+package saros.editor.internal;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.List;
+import org.apache.commons.lang3.tuple.Pair;
+import org.eclipse.jface.text.source.AnnotationModel;
+import org.eclipse.jface.text.source.IAnnotationModel;
+import org.junit.Before;
+import org.junit.Test;
+import saros.editor.annotations.ContributionAnnotation;
+import saros.net.xmpp.JID;
+import saros.session.User;
+
+public class ContributionAnnotationHistoryTest {
+  private ContributionAnnotationHistory history;
+  private AnnotationModel testModel;
+  private static final int MAX_HISTORY_LENGTH = 20;
+
+  @Before
+  public void setUp() {
+    history = new ContributionAnnotationHistory(MAX_HISTORY_LENGTH);
+    testModel = new AnnotationModel();
+  }
+
+  @Test
+  public void testAddEntriesUntilMaxLength() {
+    Pair<IAnnotationModel, List<ContributionAnnotation>> annotationsToRemove;
+
+    for (int i = 0; i < MAX_HISTORY_LENGTH; i++) {
+      annotationsToRemove = history.addNewEntry(createDummyAnnotation(testModel));
+      assertNull(annotationsToRemove);
+    }
+
+    assertEquals(MAX_HISTORY_LENGTH, history.getSize());
+  }
+
+  @Test
+  public void testaddEntriesExceedingMaxLength() {
+    Pair<IAnnotationModel, List<ContributionAnnotation>> annotationsToRemove;
+
+    final List<ContributionAnnotation> firstEntry = createDummyAnnotation(testModel);
+    annotationsToRemove = history.addNewEntry(firstEntry);
+    assertNull(annotationsToRemove);
+
+    for (int i = 1; i < MAX_HISTORY_LENGTH; i++) {
+      annotationsToRemove = history.addNewEntry(createDummyAnnotation(testModel));
+      assertNull(annotationsToRemove);
+    }
+
+    assertEquals(MAX_HISTORY_LENGTH, history.getSize());
+
+    annotationsToRemove = history.addNewEntry(createDummyAnnotation(testModel));
+    assertThat(annotationsToRemove.getRight(), is(firstEntry));
```

Comment:
Ok, this does work as `ContributionAnnotation` does not implement an equals method, meaning all comparisons are done based on object references anyways. But this could be confusing when first reading the test. If it were implementing equals, the test would not ensure the correct functionality as all dummy annotations contain the same content, meaning they would be seen as equal.

So this adjustment is not strictly necessary, but I would still suggest explicitly comparing the object references using `assertSame(...)` just to avoid such confusion (or adding a comment explaining the situation).